cmd/create.go

```diff
@@ -350,27 +337,31 @@ func (cc *createCmd) createDeployment() error {
 			deployType = cc.deployType
 		}
 
-		deployConfig, err := d.GetConfig(deployType)
+		deployConfig, err = d.GetConfig(deployType)
 		if err != nil {
 			return err
 		}
-		customInputs, err = prompts.RunPromptsFromConfigWithSkips(deployConfig, maps.Keys(flagVariablesMap))
+		for _, variable := range deployConfig.Variables {
+			fmt.Printf("Name %s, Value %s\n", variable.Name, variable.Value)
+		}
+
+		handleFlagVariables(flagVariablesMap, deployConfig)
+
+		err = prompts.RunPromptsFromConfigWithSkips(deployConfig)
 		if err != nil {
 			return err
 		}
 	}
 
-	maps.Copy(customInputs, flagVariablesMap)
-
 	if cc.templateVariableRecorder != nil {
-		for k, v := range customInputs {
-			cc.templateVariableRecorder.Record(k, v)
+		for _, variable := range deployConfig.Variables {
+			cc.templateVariableRecorder.Record(variable.Name, variable.Value)
```

Comment:
not critical for this PR, but it'd be clean to just take a `BuilderVar` for this method's arg